src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/Reliability/SqlTransientErrorDetectionStrategy.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Microsoft.Data.SqlClient.Reliability.Data;
+using Microsoft.Data.SqlClient.Reliability.Properties;
+using System.Collections.Generic;
+
+namespace Microsoft.Data.SqlClient.Reliability
+{
+    /// <summary>
+    /// Provides the transient error detection logic for transient faults that are specific to SQL Database.
+    /// </summary>
+    public sealed class TransientErrorDetectionStrategy : ITransientErrorDetectionStrategy
+    {
+        private List<int> _retriableErrors = new List<int>
```

Comment:
After the initial release, this list will become frozen forever for compatibility reasons. This is a very dangerous thing.

For example, snapshot isolation errors are missing here. Any mistake like that is going to be locked into the product forever.

I think it is a mistake to ship a concrete retry strategy. Any retry strategy is application-specific. It is not possible to create a strategy that is general enough and stands the test of time. In my opinion, this should not become part of the product.

Reply:
Based on many customer interactions, we need to have a basic concrete implementation to address all those scenarios where existing application can't be modified. The initial list is the result of various conversations within SQL organization and should be quite solid trapping connection-specific common errors, then app-specific errors can be added/removed through configuration.

Reply:
I agree, as the OP, this list is inherent to SQL Server (on-prem or Azure) and as such the product team knows best which basic scenario's are transient to the application based on the decisions made by the product team to achieve high availability.

The ask was exactly not to have these hang in 3 different external implementations, which inhibits lift & shift scenario's of existing _simple_ applications without thorough code review and changes.
Also, using 3rd party packages that use SqlClient can not be trusted to behave correctly for a lift & shift to Azure. Having these options directly in SqlClient does enable that.